tools/check_code_style.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+if [ -z "$1" ]; then
+	FILES=$(git diff master --name-only);
```

Comment:
Similar problem here.  You must check if .git/ exists.  If it does not, you may not use any git commands.  Most users do not have .git/ and this will break for the majority of users.

Reply:
@patacongo see [here](https://github.com/apache/incubator-nuttx/pull/2#discussion_r360727667)

Reply:
@davids5 we should only check the modified portion, we can't  ask contributor to fix the problem not introuced by them.

Reply:
> Similar problem here. You must check if .git/ exists. If it does not, you may not use any git commands. Most users do not have .git/ and this will break for the majority of users.

Done.

Reply:
With no workflow requirements it is not safe to assume anything.